src/hooks/useNewsletterSubscriptionModal.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react'
+import { BannerState, ShowBanner } from '../components/Banner/Banner'
+
+function useNewsletterSubscriptionModal(bannerState: BannerState) {
```

Comment:
I'm wondering if we really need this hook. 
I think the `finishSubscription` and `closeSubscriptionBanner` are closely related to the subscription banner, and could be defined in the component.

My rationale here is: what is this giving us that the component and its state can't do by themselves? 🤔

Reply:
I think the state can be handled internally.

Reply:
Today @andyesp was helping me to improve the code and we left the hook for 2 reasons:

- We must share logic between the subscription modal and the subscription banner.
- To group all those functions and variables to avoid enlarging the `index.tsx` component.

Reply:
Beautiful ✨ 🦋